src/app/navbar/navbar.component.ts

```diff
@@ -1,15 +1,38 @@
-import { Component, OnInit } from '@angular/core';
+import {Component} from '@angular/core';
 
 @Component({
-  selector: 'app-navbar',
-  templateUrl: './navbar.component.html',
-  styleUrls: ['./navbar.component.css']
+    selector: 'app-navbar',
+    templateUrl: './navbar.component.html',
+    styleUrls: ['./navbar.component.css']
 })
-export class NavbarComponent implements OnInit {
+export class NavbarComponent {
+    logo: string;
+    navBarArray: ({ link: string; icon: string; text: string })[];
```

Comment:
Można to zapisać prościej. Do określania typów w Angularze wykorzystujemy interface lub klasę ( w zależności od potrzeb).  Warto w tym samym katalogu stworzyć sobie plik np. dtos.ts, który będzie służył nam do definiowania typów w danym komponencie. W tym przypadku definicja typu będzie wyglądała następująco:

export interface Navbar {
link: string; 
icon: string; 
text: string
}

A przypisanie typu: 
navbarArray: Navbar[]